src/acm_cert_retriever/retriever.py

```diff
@@ -0,0 +1,222 @@
+    )
+    p.add(
+        "--add-downloaded-chain-to-truststore",
+        choices=["true", "false"],
+        default="false",
+        env_var="RETRIEVER_ADD_DOWNLOADED_CHAIN",
+        help="Whether or not to add the downloaded cert chain from te ARN "
+             "to the trust store",
+    )
+    p.add(
+        "--acm-key-passphrase",
+        required=True,
+        env_var="RETRIEVER_ACM_KEY_PASSPHRASE",
+        help="Passphrase to use to encrypt the downloaded key",
+    )
+    p.add(
+        "--keystore-path",
+        required=True,
+        env_var="RETRIEVER_KEYSTORE_PATH",
+        help="Filename of the keystore to save the signed keypair to",
+    )
+    p.add(
+        "--keystore-password",
+        required=True,
+        env_var="RETRIEVER_KEYSTORE_PASSWORD",
+        help="Password for the Java Keystore",
+    )
+    p.add(
+        "--private-key-alias",
+        required=True,
+        env_var="RETRIEVER_PRIVATE_KEY_ALIAS",
+        help="The alias to store the private key under in the Java KeyStore",
+    )
+    p.add(
+        "--private-key-password",
+        env_var="RETRIEVER_PRIVATE_KEY_PASSWORD",
+        help="The password used to protect ",
+    )
+    p.add(
+        "--truststore-path",
+        required=True,
+        env_var="RETRIEVER_TRUSTSTORE_PATH",
+        help="Filename of the keystore to save trusted certificates to",
+    )
+    p.add(
+        "--truststore-password",
+        required=True,
+        env_var="RETRIEVER_TRUSTSTORE_PASSWORD",
+        help="Password for the Java TrustStore",
+    )
+    p.add(
+        "--truststore-aliases",
+        required=True,
+        env_var="RETRIEVER_TRUSTSTORE_ALIASES",
+        help="Comma-separated list of aliases to use for entries in the Java TrustStore",
+    )
+    p.add(
+        "--truststore-certs",
+        required=True,
+        env_var="RETRIEVER_TRUSTSTORE_CERTS",
+        help="Comma-separated list of S3 URIs pointing at certificates to be "
+        "added to the Java TrustStore",
+    )
+    p.add(
+        "--log-level",
+        choices=["CRITICAL", "ERROR", "WARNING", "INFO", "DEBUG"],
+        default="INFO",
+        env_var="RETRIEVER_LOG_LEVEL",
+        help="Logging level",
+    )
+
+    return p.parse_args(args)
+
+
+def retrieve_key_and_cert(acm_util, rsa_util, acm_cert_arn, acm_key_passphrase):
+    """Download a key and certificate and certificate chain form ACM.
+
+    Also creates a Truststore with files from S3.
+
+    Args:
+        acm_util (Object): The boto3 utility to use
+        rsa_util (Object): The Crypto RSA utility to use
+        acm_cert_arn (String): ARN of the certificate to export
+        acm_key_passphrase (String): temporary password to use for key encryption
+
+    Returns:
+        all_data (Dict): THe json result, with the key encrypted by the passphrase
+
+    """
+    logger.info("Retrieving cert and key from AWS")
+    all_data = acm_util.export_certificate(
```

Comment:
There's no error handling around this.  The user may type in an incorrect ARN, for example, that would prevent successful cert retrieval.

Reply:
That will blow the stack afaik, but I'll test it.

Reply:
it does (blow the stack) - unit test added for that